### PR TITLE
Tidy up the PDF.js wrapper

### PR DESCRIPTION
**Pull request by linusnorton**

### CODE REVIEW

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.provider.spec.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+
+describe('PdfJsWrapperFactory', () => {
+  let component: PdfViewerComponent;
+  let fixture: ComponentFixture<PdfViewerComponent>;
+
+  beforeEach(async(() => {
+    return TestBed.configureTestingModule({
+      declarations: [ PdfViewerComponent ],
+      schemas: [NO_ERRORS_SCHEMA],
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-js/pdf-js-wrapper.provider.spec.ts]: I think we can get rid of the schemas bit, I don't think it's needed

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
   SearchResultsCount,
-  SetCurrentPageOperation, StepZoomOperation,
+  SetCurrentPageOperation,
+  StepZoomOperation,
   ZoomOperation,
   ZoomValue
 } from '../../model/viewer-operations';
 import { ToolbarToggles } from '../../model/toolbar-toggles';
+import { PdfJsWrapperFactory } from './pdf-js/pdf-js-wrapper.provider';
+import { multicast } from 'rxjs/operators';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts]: I assume this is not used

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts [hunk 3]
```diff
@@ -30,62 +33,51 @@ export class PdfViewerComponent implements AfterViewInit {
...
-    this.pdfViewer.eventBus.on('updatefindcontrolstate', e => {
-      if (e.state === FindState.NOT_FOUND || e.state === FindState.FOUND) {
-        this.searchResults.next(e.matchesCount);
-      }
-    });
-    this.pdfViewer.eventBus.on('updatefindmatchescount', e => {
-      this.searchResults.next(e.matchesCount);
-    });
-    this.pdfViewer.eventBus.on('pagechanging', e => this.currentPageChanged.next(new SetCurrentPageOperation(e.pageNumber)));
+    this.pdfViewer = this.pdfJsWrapperFactory.create(this.viewerContainer);
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts]: would it be clearer if this was called pdfWrapper

Comment (linusnorton) [reply projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.ts]: agreed

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts [hunk 4]
```diff
@@ -16,7 +14,7 @@ describe('PdfViewerComponent', () => {
       schemas: [NO_ERRORS_SCHEMA],
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts]: I don't think schemas are needed

File: projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts [hunk 5]
```diff
@@ -2,30 +2,63 @@ import { async, ComponentFixture, TestBed } from '@angular/core/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
```
Comment (benouaer) [inline projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts]: I assume this is no longer used

Comment (linusnorton) [reply projects/media-viewer/src/lib/media-viewer/viewers/pdf-viewer/pdf-viewer.component.spec.ts]: I will kill it

**Pull request merged**